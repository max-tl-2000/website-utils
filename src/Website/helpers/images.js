/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import nullish from '../../common/nullish';
import { getUrlFromCloudinary } from './cloudinary';

const getImage = (imageUrl, { width, height, radius, format = 'auto', maxDPR } = {}) => {
  let widthParam = '';
  let heightParam = '';
  let radiustParam = '';
  let formatParam = '';

  if (!nullish(width)) {
    widthParam = `w_${Math.floor(width)}`;
  }

  if (!nullish(height)) {
    heightParam = `h_${Math.floor(height)}`;
  }

  if (!nullish(radius)) {
    radiustParam = `r_${radius}`;
  }

  if (format) {
    formatParam = `f_${format}`;
  }

  return getUrlFromCloudinary(imageUrl, {
    cloudinaryParams: [
      widthParam,
      heightParam,
      radiustParam,
      formatParam,
      'c_fill',
      'q_auto:best',
      'g_auto:no_faces',
      'e_improve',
      'e_auto_brightness',
      'fl_force_strip',
    ],
    maxDPR,
  });
};

export const getImageForInventoryCarousel = (imageUrl, { height = 445, radius, ...args } = {}) => getImage(imageUrl, { height, radius, ...args });

export const getImageForPropertyCarousel = (imageUrl, { width = 1680, ...args } = {}) => getImage(imageUrl, { width, ...args });

export const getLayoutImage = (imageUrl, { height = 660, radius = 5, ...args } = {}) => getImage(imageUrl, { height, radius, ...args });

export const getUnitTypeImage = (imageUrl, { height = 277, radius = 5, ...args } = {}) => getImage(imageUrl, { height, radius, ...args });

export const getPropertyCardImage = (imageUrl, { width = 400, height = 300, ...args } = {}) => getImage(imageUrl, { width, height, ...args });

export const getCloudinaryURLForImage = (...args) => getImage(...args);
