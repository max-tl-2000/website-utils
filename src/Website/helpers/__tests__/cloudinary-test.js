/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { getCloudinaryURLForImage } from '../images';

describe('cloudinary', () => {
  it('should wrap the provided url in a cloudinary url', () => {
    const url = 'https://somepath/to/an/image?q=1';
    const result = getCloudinaryURLForImage(url);

    expect(result).toMatchSnapshot();
  });

  describe('when the url is reva api images endpoint', () => {
    it('should return the same url but with the cParams property set', () => {
      const url = 'https://reva.tech/api/images/someOtherParam?q=1';
      const result = getCloudinaryURLForImage(url);

      expect(result).toMatchSnapshot();
    });

    it('if width is set combine it with other params', () => {
      const url = 'https://reva.tech/api/images/someOtherParam?q=1';
      const result = getCloudinaryURLForImage(url, { width: 100 });

      expect(result).toMatchSnapshot();
    });
  });
});
