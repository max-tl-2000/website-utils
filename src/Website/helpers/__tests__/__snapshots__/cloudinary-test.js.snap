// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cloudinary should wrap the provided url in a cloudinary url 1`] = `"https://res.cloudinary.com/revat/image/fetch/dpr_1.0,f_auto,c_fill,q_auto:best,g_auto:no_faces,e_improve,e_auto_brightness,fl_force_strip/https%3A%2F%2Fsomepath%2Fto%2Fan%2Fimage%3Fq%3D1"`;

exports[`cloudinary when the url is reva api images endpoint if width is set combine it with other params 1`] = `"https://reva.tech/api/images/someOtherParam?cParams=dpr_1.0%2Cw_100%2Cf_auto%2Cc_fill%2Cq_auto%3Abest%2Cg_auto%3Ano_faces%2Ce_improve%2Ce_auto_brightness%2Cfl_force_strip%2F&q=1"`;

exports[`cloudinary when the url is reva api images endpoint should return the same url but with the cParams property set 1`] = `"https://reva.tech/api/images/someOtherParam?cParams=dpr_1.0%2Cf_auto%2Cc_fill%2Cq_auto%3Abest%2Cg_auto%3Ano_faces%2Ce_improve%2Ce_auto_brightness%2Cfl_force_strip%2F&q=1"`;
